pkg/mcoproducer/mcoproducer.go

```diff
@@ -0,0 +1,62 @@
+package mcoproducer
+
+import (
+	"bytes"
+	"embed"
+	"encoding/base64"
+	"fmt"
+	"text/template"
+
+	"github.com/google/go-containerregistry/pkg/name"
+)
+
+//go:embed templates
+var templateFS embed.FS
+
+func ProduceMachineConfig(machineConfigName, machineConfigPoolRef, kernelModuleImage, kernelModuleName string) (string, error) {
+	tag, err := name.NewTag(kernelModuleImage)
+	if err != nil {
+		return "", fmt.Errorf("invalid kernelModuleImage %s, input should be repo:tag format: %v", kernelModuleImage, err)
+	}
+
+	templateParams := map[string]interface{}{
+		"Image":                tag.Repository.Name(),
+		"Tag":                  tag.Identifier(),
+		"KernelModule":         kernelModuleName,
+		"MachineConfigPoolRef": machineConfigPoolRef,
+		"MachineConfigName":    machineConfigName,
+	}
+
+	replaceKernelModuleScript, err := generateTemplates(templateFS, "templates/replace-kernel-module.gotmpl", templateParams)
+	if err != nil {
+		return "", err
+	}
```

Comment:
```suggestion
        const replaceKernelModuleTmpl = "templates/replace-kernel-module.gotmpl"
	replaceKernelModuleScript, err := generateTemplates(templateFS, replaceKernelModuleTmpl, templateParams)
	if err != nil {
		return "", fmt.Errorf("couldn't generate %s template: %v", replaceKernelModuleTmpl, err)
	}
```

Reply:
i don't see the need to return a formatted error here, the returned error is pretty much explanatory

Reply:
It might be. The main benefit IMO with extending the error is that it gives you in the logs the exact last line of *our* code that fails. I find it super handy when debugging but if you insist on not having it then we can leave it as is.

Reply:
but current code will give you the exact line of code that fails, since the error in the generateTemplate function will return the precise location, explanation and error